XRTK.Oculus/Packages/com.xrtk.oculus/Editor/ValidateOculusControllerMappings.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+using UnityEngine;
+using XRTK.Oculus.Controllers;
+using XRTK.Utilities.Editor;
+
+namespace XRTK.Oculus
+{
+    [InitializeOnLoad]
+    public class ValidateOculusControllerMappings
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        static ValidateOculusControllerMappings()
+        {
+            if (!Application.isBatchMode)
+            {
+                EditorApplication.delayCall += CheckControllerMappings;
+            }
+        }
+
+        public static void CheckControllerMappings()
+        {
+            bool result = false;
+            // Test if there is an Native Oculus controller profile registered
+            if (ValidateConfiguration.ValidateControllerProfiles(new[] { typeof(OculusTouchController) }))
+            {
+                //if there is profile, also check if there are any compatible mappings.  Left tests both Left and Right
+                result = ValidateConfiguration.ValidateControllerMappings(new[] { typeof(OculusTouchController) }, XRTK.Definitions.Utilities.Handedness.Left);
```

Comment:
Why does left test both? Just curious since I can see this confusing someone returning to this code in 20 years.

Reply:
Because it is checking whether you have two separate configurations for each hand, or a single configuration for both/either hand

Reply:
Alright. As long as the comment is there it's definitely fine.